pkg/backup/restore_online_test.go

```diff
@@ -581,6 +583,65 @@ func TestOnlineRestoreErrors(t *testing.T) {
 	})
 }
 
+func TestOnlineRestoreRetryingDownloadRequests(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	defer nodelocal.ReplaceNodeLocalForTesting(t.TempDir())()
+
+	rng, seed := randutil.NewPseudoRand()
+	t.Logf("random seed: %d", seed)
+
+	alwaysFail := rng.Intn(2) == 0
+	t.Logf("always fail download requests: %t", alwaysFail)
+	totalFailures := int32(rng.Intn(maxDownloadAttempts-1) + 1)
+	var currentFailures atomic.Int32
+
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
+			Knobs: base.TestingKnobs{
+				BackupRestore: &sql.BackupRestoreTestingKnobs{
+					RunBeforeSendingDownloadSpan: func() error {
```

Comment:
I'm wondering if there's a better way to go about this instead of a testing knob? I was originally thinking of a dependency injection of the `TenantStatusServer`, but mocking the handling of `DownloadSpanRequest` would also require appropriate mocking of `SpanStatsRequest`.

Also is there an easier way to inject a `ExecutorConfig`? I see that changefeeds does so via a testing knob that replaces the `execCfg` in the DistSQL processor.